webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV2.scala

```diff
@@ -176,24 +204,52 @@ object RouteUtilV2 {
    * @param requestContext the request context.
    * @return the set of schema options submitted in the request, including default options.
    */
-  def getSchemaOptions(requestContext: RequestContext): Set[SchemaOption] =
-    Set(
-      getStandoffRendering(requestContext),
-      getJsonLDRendering(requestContext)
-    ).flatten
+  @deprecated("use getSchemaOptions() instead")
+  def getSchemaOptionsUnsafe(requestContext: RequestContext): Set[SchemaOption] =
+    getSchemaOptions(requestContext).fold(
+      errors => throw errors.head,
+      schemaOptions => schemaOptions
+    )
+
+  /**
+   * Gets the schema options submitted in the request.
+   *
+   * @param requestContext the request context.
+   * @return the set of schema options submitted in the request, including default options.
+   */
+  def getSchemaOptions(requestContext: RequestContext): Validation[Throwable, Set[SchemaOption]] =
+    for {
+      standoffRendering <- getStandoffRendering(requestContext)
+      jsonLDRendering   <- getJsonLDRendering(requestContext)
+    } yield Set(standoffRendering, jsonLDRendering).flatten
 
   /**
    * Gets the project IRI specified in a Knora-specific HTTP header.
    *
    * @param requestContext the akka-http [[RequestContext]].
    * @return the specified project IRI, or [[None]] if no project header was included in the request.
    */
-  def getProject(requestContext: RequestContext)(implicit stringFormatter: StringFormatter): Option[SmartIri] =
+  @deprecated("use getProject() instead")
+  def getProjectUnsafe(requestContext: RequestContext)(implicit stringFormatter: StringFormatter): Option[SmartIri] =
     requestContext.request.headers.find(_.lowercaseName == PROJECT_HEADER).map { header =>
       val projectIriStr = header.value
       projectIriStr.toSmartIriWithErr(throw BadRequestException(s"Invalid project IRI: $projectIriStr"))
     }
 
+  /**
+   * Gets the project IRI specified in a Knora-specific HTTP header.
+   *
+   * @param requestContext the akka-http [[RequestContext]].
+   * @return the specified project IRI, or [[None]] if no project header was included in the request.
+   */
+  def getProject(requestContext: RequestContext)(implicit stringFormatter: StringFormatter): Task[Option[SmartIri]] = {
```

Comment:
```suggestion
  def getProject(requestContext: RequestContext): ZIO[StringFormatter, Throwable, Option[SmartIri]] = ZIO.serviceWithZio[StringFormatter]{ stringformatter =>
```
Rather than depending on the `implicit` make the `ZIO` require the `StringFormatter`.